Remove non passive mode code - The Great Code Removal

morantr opened the pull request:
Since the passive mode is the only mode. Removed all non-passive mode code and features.

tomereli commented:
Great headline! Please include a code size analysis of the before and after this change to see how much you saved. We have a wiki page with the current code size analysis so you can use it as template.

morantr commented:
File | Before (07808cc8) | After (7b065eab) | Unit | Before (07808cc8) KiB | After (7b065eab) KiB | Diff Size KiB | %
-- | -- | -- | -- | -- | -- | -- | --
ncdu <prpl_repo>/build/install/bin |   |  
beerocks_controller | 2.8 | 2.6 | MiB | 2867.2 | 2662.4 | -204.8 | -7.14%
beerocks_agent | 2.6 | 2.3 | MiB | 2662.4 | 2355.2 | -307.2 | -11.54%
beerocks_cli | 500 | 480 | KiB | 500 | 480 | -20 | -4.00%
beerocks_monitor | 420 | 420 | KiB | 420 | 420 | 0 | 0.00%
ieee1905_transport | 32 | 32 | KiB | 32 | 32 | 0 | 0.00%
local_bus | 32 | 32 | KiB | 32 | 32 | 0 | 0.00%
ncdu <prpl_repo>/build/install/lib
libbtlvf.so.1.4.0 | 1.7 | 1.5 | MiB | 1740.8 | 1536 | -204.8 | -11.76%
libtlvf.so.1.4.0 | 740 | 740 | KiB | 740 | 740 | 0 | 0.00%
libbml.so.1.4.0 | 712 | 684 | KiB | 712 | 684 | -28 | -3.93%
libbcl.so.1.4.0 | 556 | 556 | KiB | 556 | 556 | 0 | 0.00%
libelpp.so.1.4.0 | 336 | 336 | KiB | 336 | 336 | 0 | 0.00%
libieee1905_transport_lib.so.1.4.0 | 204 | 204 | KiB | 204 | 204 | 0 | 0.00%
mapfcommon.so.1.4.0 | 204 | 204 | KiB | 204 | 204 | 0 | 0.00%
libbwl.so.1.4.0 | 184 | 184 | KiB | 184 | 184 | 0 | 0.00%
libbtl.so.1.4.0 | 132 | 132 | KiB | 132 | 132 | 0 | 0.00%
libbpl.so.1.4.0 | 76 | 72 | KiB | 76 | 72 | -4 | -5.26%
libieee1905_transport_messages.so.1.4.0 | 16 | 16 | KiB | 16 | 16 | 0 | 0.00%
ncdu <prpl_repo>/build/install/config
beerocks_controller.conf | 4 | 4 | KiB | 4 | 4 | 0 | 0.00%
beerocks_agent.conf | 4 | 4 | KiB | 4 | 4 | 0 | 0.00%
beerocks_controller_version | 4 | 4 | KiB | 4 | 4 | 0 | 0.00%
beerocks_agent_version | 4 | 4 | KiB | 4 | 4 | 0 | 0.00%
ncdu <prpl_repo>/build/install/share
local_bus.conf | 4 | 4 | KiB | 4 | 4 | 0 | 0.00%
logging.conf | 4 | 4 | KiB | 4 | 4 | 0 | 0.00%
prplmesh_platform_db | 4 | 4 | KiB | 4 | 4 | 0 | 0.00%
ncdu <prpl_repo>/build/install/scripts
prplmesh_utils.sh | 8 | 8 | KiB | 8 | 8 | 0 | 0.00%
ncdu <prpl_repo>/build/agent/src/beerocks/slave/CMakeFiles/beerocks_agent.dir
son_slave_thread.cpp.o | 2.4 | 2.1 | MiB | 2457.6 | 2150.4 | -307.2 | -12.50%
/platform_manager | 876 | 608 | KiB | 876 | 608 | -268 | -30.59%
/backhaul_manager | 608 | 608 | KiB | 608 | 608 | 0 | 0.00%
ap_manager_thread.cpp.o | 580 | 576 | KiB | 580 | 576 | -4 | -0.69%
beerocks_slave_main.cpp.o | 96 | 96 | KiB | 96 | 96 | 0 | 0.00%
agent_ucc_listener.cpp.o | 32 | 32 | KiB | 32 | 32 | 0 | 0.00%
ap_wlan_hal_loader.cpp.o | 24 | 24 | KiB | 24 | 24 | 0 | 0.00%
ncdu <prpl_repo>/build/agent/src/beerocks/monitor/CmakeFiles/beerocks_monitor.dir
monitor_thread.cpp.o | 348 | 348 | KiB | 348 | 348 | 0 | 0.00%
monitor_rssi.cpp.o | 132 | 132 | KiB | 132 | 132 | 0 | 0.00%
monitor_stats.cpp.o | 68 | 68 | KiB | 68 | 68 | 0 | 0.00%
monitor_db.cpp.o | 64 | 64 | KiB | 64 | 64 | 0 | 0.00%
beerocks_monitor_main.cpp.o | 56 | 56 | KiB | 56 | 56 | 0 | 0.00%
mon_wlan_hal_loader.cpp.o | 20 | 20 | KiB | 20 | 20 | 0 | 0.00%
ncdu <prpl_repo>/build/controller/src/beerocks/master/CMakeFiles/beerocks_controller.dir
/tasks | 1.9 | 1.7 | MiB | 1945.6 | 1740.8 | -204.8 | -10.53%
son_management.cpp.o | 1.1 | 1.1 | MiB | 1126.4 | 1126.4 | 0 | 0.00%
son_master_thread.cpp.o | 972 | 968 | KiB | 972 | 968 | -4 | -0.41%
/db | 800 | 788 | KiB | 800 | 788 | -12 | -1.50%
son_actions.cpp.o | 176 | 176 | KiB | 176 | 176 | 0 | 0.00%
beerocks_master_main.cpp.o | 84 | 84 | KiB | 84 | 84 | 0 | 0.00%
beerocks_master_mrouter.cpp.o | 48 | 48 | KiB | 48 | 48 | 0 | 0.00%
controller_ucc_listener.cpp.o | 44 | 44 | KiB | 44 | 44 | 0 | 0.00%
  |   |   |   |   |   |   |  
Totals: |   |   |   |   |   |   |  
Prpl Module |   |   |   | Before (07808cc8) KiB | After (7b065eab) KiB | Diff Size KiB | %
Common (ieee1905_transport,local_bus,libs,scripts,share) |   |   |   | 4984.8 | 4748 | -236.8 | -4.75%
Agent Standalone (beerocks_agent,beerocks_monitor,beerocks_agent.conf,beerocks_agent_version) + Common |   |   |   | 8075.2 | 7531.2 | -544.000000000001 | -6.74%
Controller Standalone (beerocks_controller,beerocks_controller.conf,beerocks_controller_version) + Common |   |   |   | 7860 | 7418.4 | -441.6 | -5.62%
Controller + Agent + Common |   |   |   | 20920 | 19697.6 | -1222.4 | -5.84%

[CodeSizeAnalysis.zip](https://github.com/prplfoundation/prplMesh/files/3897287/CodeSizeAnalysis.zip)

tomereli reviewed (commented):
Adam Dov must review this PR prior to approving it since it might have an effect on the standalone repeater.

tomereli commented:
> File	Before ([07808cc](https://github.com/prplfoundation/prplMesh/commit/07808cc84ceb0a829ebccd12e60230bf710b5fb5))	After ([7b065ea](https://github.com/prplfoundation/prplMesh/commit/7b065eabb24133e25995f6b30bbdfc89367f42e3))	Unit	Before ([07808cc](https://github.com/prplfoundation/prplMesh/commit/07808cc84ceb0a829ebccd12e60230bf710b5fb5)) KiB	After ([7b065ea](https://github.com/prplfoundation/prplMesh/commit/7b065eabb24133e25995f6b30bbdfc89367f42e3)) KiB	Diff Size KiB	%
> ncdu <prpl_repo>/build/install/bin	 	 					
> beerocks_controller	2.8	2.6	MiB	2867.2	2662.4	-204.8	-7.14%
> beerocks_agent	2.6	2.3	MiB	2662.4	2355.2	-307.2	-11.54%
> beerocks_cli	500	480	KiB	500	480	-20	-4.00%
> beerocks_monitor	420	420	KiB	420	420	0	0.00%
> ieee1905_transport	32	32	KiB	32	32	0	0.00%
> local_bus	32	32	KiB	32	32	0	0.00%
> ncdu <prpl_repo>/build/install/lib							
> libbtlvf.so.1.4.0	1.7	1.5	MiB	1740.8	1536	-204.8	-11.76%
> libtlvf.so.1.4.0	740	740	KiB	740	740	0	0.00%
> libbml.so.1.4.0	712	684	KiB	712	684	-28	-3.93%
> libbcl.so.1.4.0	556	556	KiB	556	556	0	0.00%
> libelpp.so.1.4.0	336	336	KiB	336	336	0	0.00%
> libieee1905_transport_lib.so.1.4.0	204	204	KiB	204	204	0	0.00%
> mapfcommon.so.1.4.0	204	204	KiB	204	204	0	0.00%
> libbwl.so.1.4.0	184	184	KiB	184	184	0	0.00%
> libbtl.so.1.4.0	132	132	KiB	132	132	0	0.00%
> libbpl.so.1.4.0	76	72	KiB	76	72	-4	-5.26%
> libieee1905_transport_messages.so.1.4.0	16	16	KiB	16	16	0	0.00%
> ncdu <prpl_repo>/build/install/config							
> beerocks_controller.conf	4	4	KiB	4	4	0	0.00%
> beerocks_agent.conf	4	4	KiB	4	4	0	0.00%
> beerocks_controller_version	4	4	KiB	4	4	0	0.00%
> beerocks_agent_version	4	4	KiB	4	4	0	0.00%
> ncdu <prpl_repo>/build/install/share							
> local_bus.conf	4	4	KiB	4	4	0	0.00%
> logging.conf	4	4	KiB	4	4	0	0.00%
> prplmesh_platform_db	4	4	KiB	4	4	0	0.00%
> ncdu <prpl_repo>/build/install/scripts							
> prplmesh_utils.sh	8	8	KiB	8	8	0	0.00%
> ncdu <prpl_repo>/build/agent/src/beerocks/slave/CMakeFiles/beerocks_agent.dir							
> son_slave_thread.cpp.o	2.4	2.1	MiB	2457.6	2150.4	-307.2	-12.50%
> /platform_manager	876	608	KiB	876	608	-268	-30.59%
> /backhaul_manager	608	608	KiB	608	608	0	0.00%
> ap_manager_thread.cpp.o	580	576	KiB	580	576	-4	-0.69%
> beerocks_slave_main.cpp.o	96	96	KiB	96	96	0	0.00%
> agent_ucc_listener.cpp.o	32	32	KiB	32	32	0	0.00%
> ap_wlan_hal_loader.cpp.o	24	24	KiB	24	24	0	0.00%
> ncdu <prpl_repo>/build/agent/src/beerocks/monitor/CmakeFiles/beerocks_monitor.dir							
> monitor_thread.cpp.o	348	348	KiB	348	348	0	0.00%
> monitor_rssi.cpp.o	132	132	KiB	132	132	0	0.00%
> monitor_stats.cpp.o	68	68	KiB	68	68	0	0.00%
> monitor_db.cpp.o	64	64	KiB	64	64	0	0.00%
> beerocks_monitor_main.cpp.o	56	56	KiB	56	56	0	0.00%
> mon_wlan_hal_loader.cpp.o	20	20	KiB	20	20	0	0.00%
> ncdu <prpl_repo>/build/controller/src/beerocks/master/CMakeFiles/beerocks_controller.dir							
> /tasks	1.9	1.7	MiB	1945.6	1740.8	-204.8	-10.53%
> son_management.cpp.o	1.1	1.1	MiB	1126.4	1126.4	0	0.00%
> son_master_thread.cpp.o	972	968	KiB	972	968	-4	-0.41%
> /db	800	788	KiB	800	788	-12	-1.50%
> son_actions.cpp.o	176	176	KiB	176	176	0	0.00%
> beerocks_master_main.cpp.o	84	84	KiB	84	84	0	0.00%
> beerocks_master_mrouter.cpp.o	48	48	KiB	48	48	0	0.00%
> controller_ucc_listener.cpp.o	44	44	KiB	44	44	0	0.00%
>  	 	 	 	 	 	 	 
> Totals:	 	 	 	 	 	 	 
> Prpl Module	 	 	 	Before ([07808cc](https://github.com/prplfoundation/prplMesh/commit/07808cc84ceb0a829ebccd12e60230bf710b5fb5)) KiB	After ([7b065ea](https://github.com/prplfoundation/prplMesh/commit/7b065eabb24133e25995f6b30bbdfc89367f42e3)) KiB	Diff Size KiB	%
> Common (ieee1905_transport,local_bus,libs,scripts,share)				4984.8	4748	-236.8	-4.75%
> Agent Standalone (beerocks_agent,beerocks_monitor,beerocks_agent.conf,beerocks_agent_version) + Common				8075.2	7531.2	-544.000000000001	-6.74%
> Controller Standalone (beerocks_controller,beerocks_controller.conf,beerocks_controller_version) + Common				7860	7418.4	-441.6	-5.62%
> Controller + Agent + Common	 	 	 	20920	19697.6	-1222.4	-5.84%
> [CodeSizeAnalysis.zip](https://github.com/prplfoundation/prplMesh/files/3897287/CodeSizeAnalysis.zip)

@morantr Great analysis - kudos on the CodeSizeAnalysis.ods which is perfectly clear and organized.
Actually, @arnout what do you think about including the CodeSizeAnalysis.ods in our documentation folder?

tomereli commented on framework/platform/bpl/include/bpl/bpl_err.h:
```diff
@@ -26,74 +26,54 @@ extern "C" {
+/* 37 */ ERROR_CODE(BPL_ERR_CONFIG_NO_VALID_BACKHAUL_INTERFACE)                                     \
+/* 38 */ ERROR_CODE(BPL_ERR_WATCHDOG_PROCESS_STUCK)                                                 \
+/* 39 */ ERROR_CODE(BPL_ERR_WATCHDOG_PROCESS_ZOMBIE)                                                \
+/* 40 */ ERROR_CODE(BPL_ERR_UNKNOWN)                                                                \
```
```
/* 40 */ ERROR_CODE(BPL_ERR_UNKNOWN)                                                                \
/* 41 */ ERROR_CODE(BPL_ERR_OPERATION_NOT_SUPPORTED)  
```
Have to be 1 and 2 respectively.

morantr replied:
why? to make it 1 and 2 I need to change all other numbers 😱

tomereli replied:
So do it or remove the numbers :/ 

No error is normally 0
Other errors start with 1,2,3,...
It makes sence that unknown error would be first or last. The OPERATION_NOT_SUPPORTED should also be one of the first ones or the last ones IMO.
But the real reason is that these are used in other locations without actual using the enum like I explained before - you can fix that or just keep the ordering like I suggested..

morantr replied:
Add a fixup. I kept the numbering since it is helpful when debugging since the error notification prints a number.

tomereli commented on agent/src/beerocks/slave/ap_manager_thread.cpp:
```diff
@@ -337,8 +337,6 @@ void ap_manager_thread::after_select(bool timeout)
             }
 
             // Set the time for the first radio info polling
-            next_tx_polling_timestamp =
```
commit message typo "has been removed fro the ap manager" --> "has been removed **from** the ap manager"

morantr replied:
I'll fix it on the next rebase. Leaving it unresolved for now so I won't forget.

adam1985d commented on framework/CMakeLists.txt:
```diff
@@ -38,14 +38,9 @@ include(multiap-helpers)
 set(MSGLIB "zmq" CACHE STRING "Which messaging library backend to use")
 set_property(CACHE MSGLIB PROPERTY STRINGS "zmq" "nng" "None")
-if(PASSIVE_MODE)
-message(STATUS "Force MSGLIB=None in passive mode")
-set(MSGLIB "None")
```
these 2 lines inside the passive_mode condition were performed always
this is a change to the passive_mode and not removal on non-passive

morantr replied:
Yes, it is not relevant so I removed it. On the next single code base PR, I'll set the MSGLIB to NONE on RDKB and instead of PASSIVE_MODE.

adam1985d commented on framework/platform/bpl/uci/cfg/bpl_cfg_extra.cpp:
```diff
@@ -171,12 +147,6 @@ int bpl_cfg_set_wifi_radio_tx_state(const char iface[BPL_IFNAME_LEN], int enable
 
 #else //PASSIVE_MODE is on
```
you can remove the comment

and what "if" is this else referring to? shouldn't it also be removed?

arnout replied:
> you can remove the comment
> 
> and what "if" is this else referring to? shouldn't it also be removed?

It's removed in a later commit.

arnout commented:
> Actually, @arnout what do you think about including the CodeSizeAnalysis.ods in our documentation folder?

Ideally, I'd like something that can easily be updated by CI. But for the time being that .ods is a good idea.

arnout commented on framework/platform/bpl/uci/cfg/bpl_cfg_helper.cpp:
```diff
@@ -13,263 +13,6 @@
 
 using namespace beerocks::bpl;
 
-#ifndef BPL_PASSIVE_MODE
```
The removal of this ifndef is done in commit "bpl: Remove "advertise_ssid" setter from bpl" but it has nothing to do with that. So it should be a separate commit, or part of "framework: Remove passive mode from framework and bpl CmakeList files".

(nitpick)

morantr replied:
Oops, I guess I mistakenly `git add` it to the commit. Since it is a nitpick I guess I would leave it...

arnout commented on framework/platform/bpl/uci/cfg/bpl_cfg.cpp:
```diff
@@ -173,7 +173,6 @@ int bpl_cfg_get_wifi_params(const char iface[BPL_IFNAME_LEN], struct BPL_WLAN_PA
         wlan_params->enabled = !disabled;
     }
     retVal |= bpl_cfg_get_channel(index, &wlan_params->channel);
-    retVal |= bpl_cfg_get_ssid_advertisement_enabled(index, &wlan_params->advertise_ssid);
```
The definition of this function bpl_cfg_get_ssid_advertisement_enabled should be removed as well.

morantr replied:
Good catch! thanks.

arnout commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -207,9 +207,7 @@ void slave_thread::on_thread_stop() { stop_slave_thread(); }
-    if (slave_state == STATE_WAIT_FOR_WIFI_CONFIGURATION_UPDATE_COMPLETE ||
-        slave_state == STATE_WAIT_FOR_ANOTHER_WIFI_CONFIGURATION_UPDATE ||
-        slave_state == STATE_WAIT_FOR_UNIFY_WIFI_CREDENTIALS_RESPONSE) {
+    if (configuration_in_progress) {
```
This change is a bit different than "removing unused stuff", so it would have been nicer to review in a separate commit (nitpick).

morantr replied:
I had needed to remove the first two comparisons as part of this commit and replace it with the `configuration_on_process` but since I knew the 3rd comparisons would need to drop as well, I preferred doing it all together.

arnout commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -3452,63 +3381,10 @@ bool slave_thread::slave_fsm(bool &call_slave_select)
-            slave_state = STATE_START_MONITOR;
-        }
+        LOG(TRACE) << "goto STATE_START_MONITOR";
+        slave_state = STATE_START_MONITOR;
```
Since we go to this state unconditionally, we can remove one of these two states. I think it's best to keep STATE_START_MONITOR. Do this in a separate commit please.

morantr replied:
done.

arnout commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -3440,6 +3315,10 @@ bool slave_thread::slave_fsm(bool &call_slave_select)
         if (!platform_settings.local_gw) {
             // Wireless config
+
+            // TODO: On passive mode, mem_only_psk is always be set, so supplying the credentials
+            // to the backhaul manager will no longer be necessary, and therefore should be be
+            // removed completely from beerocks including the BPL.
```
I don't think this is true. `mem_only_psk` is an option of wpa_supplicant that disables the writing out of the psk when wpa_supplicant is asked to write out its config file. It's a fairly useless option IMO - we anyway have to write out the PSK to be able to re-establish the backhaul link after reboot!

tomereli replied:
This is part of the Intel SDL, `mem_only_psk` is not useless, according to my understanding when it is set, the supplicant will wait to get the PSK some platform entity so it is never stored unencrypted in a file. This is the only mode allowed according to the SDL. @morantr should provide more details though.

morantr replied:
@tomereli do I need to edit my comment?
If yes, please write me down exactly what to write.

arnout commented on controller/src/beerocks/bml/internal/bml_internal.cpp:
```diff
@@ -1514,158 +1498,88 @@ int bml_internal::set_wifi_credentials(const std::string ssid, const std::string
+    if (!is_local_master()) {
+        LOG(ERROR) << "Command supported only on local master!";
+        // Clear the promise holder
+        m_prmWiFiCredentialsUpdate = nullptr;
```
It would be better to move this check to before the promise is created.
However, since I believe this entire function should be eliminated: nitpick.

morantr replied:
I didn't change it. Only remove the part that sends this message directly to the agent. For some reason, the git diff looks like I changed it. Anyway, this should not be removed since it is updating the controller about new credentials, which will eventually trigger auto-config.

arnout commented on agent/src/beerocks/slave/platform_manager/platform_manager_thread.cpp:
```diff
@@ -988,135 +876,6 @@ bool main_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
-            bool success = true;
-
-            bool is_iface_an_ap_iface = false;
-            if (ap_ifaces.find(iface) != ap_ifaces.end()) {
```
This makes me realize:

- ap_ifaces is only used to determine if an interface in bpl_iface_status_map is an AP;
- bpl_iface_status_map is only used in platform_operational_state_check() and to trigger bpl_cfg_notify_iface_status();
- bpl_cfg_notify_iface_status() actually doesn't do anything in any implementation;
- bpl_iface_status is only used for bpl_cfg_notify_iface_status;
- BPL_INTERFACE_STATUS_NOTIFICATION is only used for bpl_cfg_notify_iface_status;
- enable_bpl_iface_status_notifications is only used for bpl_cfg_notify_iface_status.

There are a lot of things that can be removed here...

In fact, I think the whole PLATFORM_WIFI_INTERFACE_STATUS_NOTIFICATION -> PLATFORM_OPERATIONAL_NOTIFICATION back-and-forth can be removed. As far as I can see, in passive mode platform operational state does not or should not depend on wifi status.

morantr replied:
@tomereli to remove the platform status as well?
From what I remember, we started it in the first place in order to set LEDs on the platform.

arnout commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -2562,17 +2365,6 @@ bool slave_thread::handle_cmdu_monitor_message(
             iface_status_operational_state = false;
         }
 
-        if (slave_state == STATE_OPERATIONAL && notification_in->new_tx_state() == 0 &&
-            notification_in->new_hostap_enabled_state() == 1) {
```
We could now merge new_tx_state and new_hostap_enabled_state into a single boolean, but that's possibly a bit of overkill.

morantr replied:
Basically for operational status to the platform manager, we had defined that the front radio should be enabled, and the should be TXing as well. If we remove the status notification, this could be removed as well.
@tomereli, need your call here.

tomereli replied:
different PR I think.

arnout commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -2562,17 +2365,6 @@ bool slave_thread::handle_cmdu_monitor_message(
-            notification_in->new_hostap_enabled_state() == 1) {
-            if (!set_wifi_iface_state(config.hostap_iface, WIFI_IFACE_OPER_ENABLE)) {
-                LOG(ERROR) << "error enabling hostap tx --> slave_reset();";
-                platform_notify_error(BPL_ERR_SLAVE_IFACE_CHANGE_STATE_FAILED,
-                                      config.hostap_iface.c_str());
```
Actually, bpl_cfg_notify_error() is not implemented in any bpl implementation, so maybe all that stuff can be removed as well.

morantr replied:
It was implemented on UGW bpl if I remember correctly. Who removed it?
@tomereli what to do with it? If we remove it, I will need to drop the commit that touched the `bpl_err.h` and remove this file.

tomereli replied:
different PR

arnout commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -3115,120 +2907,21 @@ bool slave_thread::slave_fsm(bool &call_slave_select)
-                                  config.hostap_iface.c_str());
-            stop_on_failure_attempts--;
-            slave_reset();
             break;
```
Personally, I don't like the "break out of switch" pattern very much. It's OK-ish within a loop, but I find that breaking out of a switch makes the code flow harder to follow. So I'd have put (or rather, kept) an else here instead. (nitpick)

morantr replied:
I think otherwise, `break` statements help understand that you need to continue reading the code. Also, it can turn the code less nested.

arnout commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -217,23 +217,6 @@ bool slave_thread::socket_disconnected(Socket *sd)
 
-    auto ap_manager_err_code_to_bpl_err_code = [](int code) -> int {
-        if (code == ap_manager_thread::eThreadErrors::APMANAGER_THREAD_ERROR_HOSTAP_DISABLED) {
-            return BPL_ERR_AP_MANAGER_HOSTAP_DISABLED;
```
These values are no longer used so should be removed from bpl_err.h as well.

morantr replied:
I know, I decided to keep them because I thought we will want to notify about it.
Anyhow, if @tomer will decide to remove the bpl_notification they will be removed anyway.

arnout commented on common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.cpp:
```diff
@@ -215,36 +213,19 @@ bool base_wlan_hal_dwpal::fsm_setup()
+                            LOG(ERROR) << "ap_enabled with tx OFF!";
                             error = true;
                         }
+                    } else if (m_radio_info.wifi_ctrl_enabled != 0) {
```
Maybe we should change wifi_ctrl_enabled into a bool parameter then? (nitpick)

tomereli replied:
We should change its name. It is basically `hostapd_cli -iwlanX status`.. the name is super confusing. We used it for the `ACS_DONE` patch in hostapd in which case `wifi_ctrl_enabled=2` after hostapd init complete, and beerocks sent `chann_switch` command which triggered hostapd init completion. The whole point was to make sure no clients ever connect before beerocks was attached, so we won't miss connected clients.
In passive mode we simply don't need this anymore (we get the connected clients in a different way).

Bottom line - we should change this to `bool hostap_enabled` and I don't think it should be a (nitpick), but a requirement. But this can be done in a separate PR..

morantr replied:
I kept it and integer because I didn't know if the hostapd is still able to send `2`.
So if it did, I thought that it would be helpful to print it for debugging purposes.

arnout commented on agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp:
```diff
@@ -590,23 +590,9 @@ bool main_thread::backhaul_fsm_main(bool &skip_select)
                 m_sConfig.eType = SBackhaulConfig::EType::Wired;
 
-                LOG(DEBUG) << "add wired_backhaul to bridge " << m_sConfig.wire_iface;
-                if (!network_utils::linux_add_iface_to_bridge(m_sConfig.bridge_iface,
```
This function can now also be removed from network_utils.

However, we may need to revive this code for backhaul optimisation (wired/wireless) so perhaps it's better to keep it in network_utils for the time being.

tomereli replied:
This should not be simply removed but replaced with a check - and trigger slave reset if the wired interface is not in the bridge IMO.

tomereli replied:
In addition, wired_backhaul option in configuration file used to mark that we either use the WAN interface for wired backhaul (if enabled) OR as a normal LAN interface. Since we are in passive mode, this should be taken care of by the platform, so this option should also be removed.

adam1985d replied:
> This should not be simply removed but replaced with a check - and trigger slave reset if the wired interface is not in the bridge IMO.

if this is the approach taken - it means the check of non-existence of the interface in bridge should be done when "remove_from_bridge" was performed

morantr replied:
Added check if the iface is in the bridge.

morantr replied:
I will remove `wired_backhaul` option from bpl in a different commit.

arnout commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -1593,49 +1593,6 @@ bool slave_thread::handle_cmdu_platform_manager_message(
-            // response->config_start == 1, but prev conf change is not finished yet
-            slave_state_timer =
-                std::chrono::steady_clock::now() +
-                std::chrono::seconds(beerocks::SON_SLAVE_WAIT_AFTER_WIFI_CONFIG_UPDATE_SEC);
```
SON_SLAVE_WAIT_AFTER_WIFI_CONFIG_UPDATE_SEC is no longer used so should be removed.

morantr replied:
removed.

arnout commented on controller/src/beerocks/master/tasks/channel_selection_task.cpp:
```diff
@@ -1095,57 +1083,6 @@ void channel_selection_task::work()
-    case eState::SEND_HOSTAP_TX_ON_REQUEST: {
-
-        Socket *hostap_mac_sd = database.get_node_socket(hostap_mac);
-        auto request =
-            message_com::create_vs_message<beerocks_message::cACTION_CONTROL_HOSTAP_TX_ON_REQUEST>(
-                cmdu_tx);
```
I notice that the monitor still has the concept of tx_enabled. Is that still relevant?

morantr replied:
The monitor notifies about TX change to the son_slave which influences the operational notification to the platform. If we remove the iface status notification we could remove it as well.

arnout reviewed (approved):
Mostly I identified other things that can be removed. Only two of them really belong in this PR, the rest is more for future reference.

There are still a large number of fapi references in the dummy bwl - those should probably be removed as well.

Can set_wifi_credentials() be removed from BML?
This would also remove action WIFI_CREDENTIALS_UPDATE which is anyway not implemented.

To be discussed: what to do with backhaul manager.
Backhaul link is supposed to be up when prplMesh is started.
If it is not, we should probably wait for it to come up.
At some point we'll have to do backhaul management, e.g. switch wired/wireless.
Also we will need to be able to update credentials, but not for R2 or R3.
However, currently backhaul credentials are retrieved from platform settings.
That is definitely wrong.
Eliminating that allows us to eliminate bpl_cfg_get_beerocks_credentials and a lot of related code.

The PR can be merged after rebase and removing the two little things I mentioned, the rest can be done in follow-up. So approving. Great work!

tomereli commented:
> Can set_wifi_credentials() be removed from BML?
> This would also remove action WIFI_CREDENTIALS_UPDATE which is anyway not implemented.

It should be replaced with something like `trigger_autoconfig_renew()` which will only send autoconfig renew too all connected agents.

tomereli reviewed (approved):
I have some important changes I'd like to request for this PR, see my comments. However, nothing blocking, so you could do this in a separate PR.
So as I see it - there are 2 options - 
1. Fix all remaining comments including nitpicks as part of this PR
2. Open a task - 'Code removal finalize' or something like that, and refer to this PR from it, and we'll get to it sometime.

Since you are 100% occupied with the single codebase tasks at the moment, lets go with option 2 - so approving.

arnout commented:
> > Can set_wifi_credentials() be removed from BML?
> > This would also remove action WIFI_CREDENTIALS_UPDATE which is anyway not implemented.
> 
> It should be replaced with something like `trigger_autoconfig_renew()` which will only send autoconfig renew to all connected agents.

NACK. I think there should instead be a way to set the AP configuration (like, DEV_SET_CONFIG) and calling that will implicitly do a renew after setting the new config in the database. I don't see a use case for changing the configuration without pushing it to the agents.

adam1985d commented on common/beerocks/tlvf/yaml/beerocks/tlvf/beerocks_message_action.yaml:
```diff
@@ -178,9 +178,6 @@ eActionOp_PLATFORM:
 
   ACTION_PLATFORM_WIFI_INTERFACE_STATUS_NOTIFICATION: 55
 
-  ACTION_PLATFORM_WIFI_CONFIGURATION_UPDATE_REQUEST: 56
```
maybe after all this refactoring is complete we need a task to review the enum numbering and reorder them?
@tomereli

tomereli replied:
Agree - please create an issue for it :)

tomereli commented:
> > > Can set_wifi_credentials() be removed from BML?
> > > This would also remove action WIFI_CREDENTIALS_UPDATE which is anyway not implemented.
> > 
> > 
> > It should be replaced with something like `trigger_autoconfig_renew()` which will only send autoconfig renew to all connected agents.
> 
> NACK. I think there should instead be a way to set the AP configuration (like, DEV_SET_CONFIG) and calling that will implicitly do a renew after setting the new config in the database. I don't see a use case for changing the configuration without pushing it to the agents.

Even better, ACK :)

adam1985d commented on controller/src/beerocks/bml/internal/bml_internal.h:
```diff
@@ -215,19 +206,18 @@ class bml_internal : public beerocks::socket_thread {
+    beerocks::promise<bool> *m_prmGetVapListCreds       = nullptr;
+    beerocks::promise<bool> *m_prmSetVapListCreds       = nullptr;
+    beerocks::promise<bool> *m_prmOnboard               = nullptr;
+    beerocks::promise<bool> *m_prmWiFiCredentialsUpdate = nullptr;
```
is the credentials update still required?
@morantr

arnout replied:
> is the credentials update still required?

I think we need a credentials update BML command that is the equivalent of DEV_SET_CONFIG, and that triggers an autoconfig renew. So, a single command that sets *all* credentials.

morantr replied:
What Arnout said...

adam1985d reviewed (approved):
from the repeater POV I don't see any major affect

note that any additional changes/fixups to backhaul_manager_thread.cpp in general and specifically wireless connection flow needs to be reviewed

arnout commented:
>     2. Open a task - 'Code removal finalize' or something like that, and refer to this PR from it, and we'll get to it sometime.

#532

morantr committed:
framework: Remove passive mode from framework and bpl CmakeList files

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
bpl: Remove "advertise_ssid" setter from bpl

"advertise_ssid" is a flag that in the passive mode code part does not
exist and not implemented. Also, it is not being used at all.
Currently, this flag is saved on the controller database, but it has no
use.

Remove this flag from bpl, and any trace of it on tlvf, agent and
controller.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
bpl: Remove "set_wifi_credentials" from bpl

On passive mode "set_wifi_credentials" is not relevant.
Therefore, remove it from the bpl and all related code which are using
it, including credentials unify flow on the son_slave, and it's related
messages on tlvf, son_slave configuration and cmake file.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
bpl: Remove "set_beerocks_credentials" from bpl

On passive mode "set_beerocks_credentials" is not relevant.
Therefore, remove it from the BPL and all related code which are using
it, including the bml_wifi_credential_supdate_task, and it's related
messages on TLVF, as well the ability to change the credentials from
BML directly to the platform manager without the controller involved.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
bpl: Remove interface and radio operations from BPL

On passive mode changing the  interface and radio states is not
relevant.
Therefore, remove the functions:
bpl_cfg_set_wifi_iface_state
bpl_cfg_set_wifi_radio_tx_state

Somewhere along the way, "bpl_cfg_set_wifi_iface_state" was replaced
by the folowing functions which were removed as well:
bpl_wlan_ap_start
bpl_wlan_ap_stop
bpl_wlan_sta_start
bpl_wlan_sta_stop
bpl_wlan_restore

The following functions have been removed too since their names imply
on wifi operations, and they didn't have any use in the code:
bpl_wlan_ap_enable
bpl_wlan_ap_disable
bpl_wlan_sta_enable
bpl_wlan_sta_disable
bpl_wlan_sta_acs

Since the following functions were involved in states which depended on
changing interfaces states the were removed also:
bpl_wlan_ready
bpl_wlan_ap_postinit

All the code that has been related or using the functions mention in
this commit was removed. It includes TLVF messages, the ability to change
tx or interfaces state from the beerocks_agent optional arguments, and
states on the son_slave.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
bpl: Remove passive_mode from BPL

Since the passive mode is the only mode. Removed "passive_mode" getters
from BPL and from the UCI.
All the code that had been executed on the non-passive move in the
agent, controller and common (BWL) has been removed.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
agent: Remove Linux calls to add/remove an interface from the bridge

On passive mode which is the only mode, we are not allowed to add/remove
interface from the bridge.
Removed from the backhaul manager, the code which does it.
Instead, add a check to if the interface is in the bridge.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
tools: Remove PASSIVE_MODE option from Klocwork script

Since passive mode is the only mode, removed PASSIVE_MODE option from
the klocwork script.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
bml: Remove wifi configuration update request

On passive mode, setting the platform manager on wifi configuration mode
is no longer needed.
Removed from the BML the ability to send to the platform manager wifi
configuration update request which sets the agent on configuration state.
Removed also, all related messages, and related code on the son_slave
and the platform manager.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
bpl: Removed unused bpl error codes

Removed unused BPL error code from bpl_err.h.
A lot of the error which has been removed is related to non-passive
mode code cleanups.
As well aligned the code which was messed up by clang-format.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
agent: Remove unused code from ap_mamanger

In the past, the ap manager monitored the tx state of the AP.
This ability has been removed from the ap manager, but there are some
leftovers of it in the code. Removed it.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
tlvf: Remove TX_ON/OFF request messages.

In the past, we could change the tx state of the AP on the
ap manager.
This option has been removed, but some leftover was remained in
the code.
Removed TX_ON/OFF request messages and all related code that has been
using it.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
bpl/bwl: Remove BRCM code

Remove BRCM code from bwl and from bpl.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
bwl: Remove BWL FAPI from the code

Remove BWL FAPI from the code. Also removed any reference to the FAPI
in the code and replace it with the word "bwl".
(GOOD RIDDANCE).

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
agent: Remove useless STATE_WAIT_FOR_AP_MANAGER_INIT_DONE_NOTIFICATION

This state does nothing. Go immediately to STATE_WAIT_FOR_AP_MANAGER_JOINED.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
bpl: Remove "wired_backhaul" from the BPL

wired_backhaul option decides whether or not to treat the WAN interface
as LAN if it is disabled, or as WAN if it is enabled. We always use the
WAN as WAN, so the code should behave always as if
wired_backhaul='true'.

Removed "wired_backhaul" getter from the BPL and all of its related
code.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>
